In [1]:
import dataset, model, visualise, evaluate

Initializing the data manager
Data manager initialization finished


In [2]:
data_set = dataset.load_data_set('../res/cache/datasets/dataset_522') 

In [4]:
from tqdm import tqdm
clip_dists = [ model.compute_clip_regions_dst(dp) for dp in tqdm(data_set)] 

  0%|          | 0/522 [00:00<?, ?it/s]C:\Users\H492635\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\nn\functional.py:5476: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:263.)
  attn_output = scaled_dot_product_attention(q, k, v, attn_mask, dropout_p, is_causal)
100%|██████████| 522/522 [00:57<00:00,  9.13it/s]


In [21]:
adjusted_clip_dists = [evaluate.adjust_region_score(clip_res, len_max=5, len_penalty=0.0000001, dist_far_threshold=0.8, dist_far_penalty=0.0001, dist_near_threshold=1000) for clip_res in clip_dists]


In [10]:
clip_dists

[{<annotation.ImageAnnotation at 0x1bde2b390>: [(<parse_xml.TextRegion at 0x83e3d90>,
    0.9996519088745117),
   (<parse_xml.TextRegion at 0x83e1f50>, 0.00034329836489632726),
   (<parse_xml.TextRegion at 0x1bde2a050>, 3.582639919841313e-06),
   (<parse_xml.TextRegion at 0x83e3110>, 1.1814291838163626e-06),
   (<parse_xml.TextRegion at 0x83e2110>, 2.321486292089503e-08)]},
 {<annotation.ImageAnnotation at 0x1bdea2990>: [(<parse_xml.TextRegion at 0x1bdea1310>,
    0.9646142721176147),
   (<parse_xml.TextRegion at 0x1bdea1410>, 0.030526746064424515),
   (<parse_xml.TextRegion at 0x1bdea1f90>, 0.004830033052712679),
   (<parse_xml.TextRegion at 0x1bdea1690>, 2.8945685699000023e-05)]},
 {<annotation.ImageAnnotation at 0x1bde81190>: [(<parse_xml.TextRegion at 0x1bdea2e90>,
    0.9999891519546509),
   (<parse_xml.TextRegion at 0x1bdea3210>, 1.045154294843087e-05),
   (<parse_xml.TextRegion at 0x1bdea2f90>, 2.273248327355759e-07),
   (<parse_xml.TextRegion at 0x1bdea3cd0>, 7.49637791841451e-

In [26]:
import importlib
importlib.reload(evaluate)

<module 'evaluate' from 'C:\\Users\\H492635\\PycharmProjects\\KNN\\scripts\\evaluate.py'>

In [27]:
results = evaluate.eval_region_data_set(data_set, adjusted_clip_dists)

invalid=4


In [28]:
evaluate.eval_region_data_set(data_set, clip_dists), evaluate.calculate_metrics(*evaluate.eval_region_data_set(data_set, clip_dists)), results, evaluate.calculate_metrics(*results)
# adjusted_clip_dists (0.6471518987341772, 0.7602230483271375, 0.6991452991452991))

invalid=4
invalid=4


(array([ 343,  324, 3518,  195]),
 (0.5142428785607196, 0.637546468401487, 0.5692946058091286),
 array([ 409,  223, 3619,  129]),
 (0.6471518987341772, 0.7602230483271375, 0.6991452991452991))

# Results:
- text_lines.len > 5 penalty: `0.0000001`
    - dataset 158 regions no dist: `((0.5822784810126582, 0.4742268041237113, 0.5227272727272727), array([  92,   66, 1003,  102]))`
    - dataset 522 regions no dist: `((0.5957854406130269, 0.5371329879101899, 0.5649409627611262), array([ 311,  211, 3782,  268]))`
    - dataset 158 regions dist(negative penalty: 0.0001, positive penalty: 10000.0): `((0.689873417721519, 0.5618556701030928, 0.6193181818181819), array([ 109,   49, 1020,   85]))`
    - dataset 522 regions dist(len penalty: inf, negative penalty: 0.0001, positive penalty: 10000.0): `array([ 409,  223, 3619,  129]), (0.6471518987341772, 0.7602230483271375, 0.6991452991452991))`